gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/SubscriptionServiceImpl.java

```diff
@@ -509,6 +509,17 @@ public SubscriptionEntity process(ProcessSubscriptionEntity processSubscription,
                 subscription.setClosedAt(new Date());
             }
 
+            if (planEntity.getSecurity() == PlanSecurityType.API_KEY && subscription.getStatus() == Subscription.Status.ACCEPTED) {
+                if (StringUtils.isNotEmpty(processSubscription.getCustomApiKey())) {
+                    if (apiKeyService.exists(processSubscription.getCustomApiKey())) {
+                        throw new ApiKeyAlreadyActivatedException("The API key is already activated");
```

Comment:
```suggestion
                        throw new ApiKeyAlreadyActivatedException("This API key is already used by another subscription");
```
Could be a more user-friendly wording, improving comprehensibility